server/gfmd/auth_uid.c

```diff
@@ -171,7 +171,9 @@ auth_uid_to_global_username_server_auth(void *closure,
 	gfarm_error_t (*get_hostname)(enum gfarm_auth_id_role, const char *,
 	    char **hostnamep),
 	enum gfarm_auth_id_role *auth_user_id_rolep,
-	char **global_usernamep, const char *diag)
+	char **global_usernamep,
+	struct user *(*user_lookup_by_auth_user_id)(const char *),
```

Comment:
This is good.

However, there is one problem that it violates the coding conventions listed in the following URL:
https://github.com/oss-tsukuba/gfarm/blob/4ff8b636abb697970e5c1d4188aa3ecf9f3baa7b/doc/internal/ja/gfarm-coding-style.txt#L53

Please move the user_lookup_by_auth_user_id parameter from the 5th parameter to the 2nd parameter to conform to the coding convention.